Source/Diagnostics/WarpXOpenPMD.cpp

```diff
@@ -1166,10 +1177,44 @@ WarpXOpenPMDPlot::GetMeshCompNames (int meshLevel,
     }
 
     if ( 0 == meshLevel )
-      return;
+        return;
 
     field_name += std::string("_lvl").append(std::to_string(meshLevel));
 }
+/* Split varname into name and theta mode mode index
+ *
+ * @param varname [IN]: variable name
+ * @param varname field_str_end_index [OUT] : if varname = fieldName_modeNumber_realOrImag,  index of underscore after fieldName, otherwise, varname.size()
+ * @param varname mode_index [OUT] : if varname = fieldName_modeNumber_realOrImag, returns 2 * modeNumber - 1 + (realOrImag == 'imag'), otherwise, -1
+ */
+std::tuple<int, int>
+GetFieldModeIndices (const std::string& varname)
+{
+    int mode_index = -1;
+    int field_str_end_index = varname.size();
+    if (varname.size() >= 6) {
+        std::string real_imag = varname.substr(field_str_end_index - 4);
```

Comment:
I think this would not work for >5 modes

Reply:
It works as far as I can test

Reply:
irrelevant as a new version was implemented